packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -539,7 +539,7 @@ def self.remove_flag_for_key(config, flag, key)
 
     def self.add_search_path_if_not_included(current_search_paths, new_search_path)
         if !current_search_paths.include?(new_search_path)
```

Comment:
```suggestion
        entry = " #{new_search_path}"
        if !current_search_paths.include?(new_search_path) && !current_search_paths.include?(entry)
          current_search_paths << entry
```

Reply:
@cipolleschi Thanks for pointing this out! I didn't realise the input could be both string or array. 

I updated the code a bit different than your suggestion. I didn't really like the solution where we depend on extra spaces to be added. In stead, I now strip them if they exist, resulting in a cleaner project without unnecessary spaces.

Do you think it's okay like this? Or is there are reason to keep the "extra spaces" solution?

I also added some tests for the function.

You can run them using by running: `ruby packages/react-native/scripts/cocoapods/__tests__/utils-test.rb -n "/add_search_path_if_not_included/" --verbose`

Results:

```
Loaded suite packages/react-native/scripts/cocoapods/__tests__/utils-test
Started
UtilsTests:
  test_add_search_path_if_not_included_with_array_input:								.: (0.000148)
  test_add_search_path_if_not_included_with_invalid_input:								.: (0.000059)
  test_add_search_path_if_not_included_with_string_input:								.: (0.000040)

Finished in 0.00047 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------
3 tests, 7 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------
6382.98 tests/s, 14893.62 assertions/s
```

Reply:
I thought of a possible issue: folder names with spaces. Original code preserves these, my code did't. Updated function and tests to support folders with spaces.